### ISSUE TITLE
ReflectionExtension should not warn about static::class

### ISSUE THREAD

**Issue by marcospassos** [pepakriz/phpstan-exception-rules#62]
Although safe, the rule reports the following snippet as missing `@throws`:

```php
new ReflectionClass(static::class);  // Missing @throws ReflectionException
```
The `ReflectionClass` should handle this case without reporting warnings.

**Comment by pepakriz** [comment]
I think It is because of `self::class` is resolved as ConstantStringType, but `static::class` is resolved as StringType. So there is no guarantee that the string contains a class name.

**Comment by pepakriz** [comment]
WDYT about https://github.com/pepakriz/phpstan-exception-rules/compare/static-workaround?expand=1 ? Currently there is no better way than this ugly hack :(
